### PR TITLE
fix: back funtionality of the app

### LINKED ISSUE #726: back stack issue in ticket details
**Describe the bug**

app goes in background if you press the back button in ticket details or back option on top menu , back stack also produces problem when you access ticket detail through likes window(Likes->selecta ny event->tickets->press back button)

**To Reproduce**
1. Select any event 
2. Select Tickets
3. Select BackArrow on top Menu(or press back button)

In likes pannel
1.go to likes pannel
2. Select any event 
3. Select Tickets
4. Select BackArrow on top Menu(or press back button)

**Expected behavior**
should go back to event details pannel from which we have selected tickets option

**Smartphone Info:**

|Device    ->redmi note 4
|Android Version| ->7.0

**Would you like to work on the issue?**

- [ x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by pranavpandey1998official**
Fixes #726 

the issue was in onBackPresses in main activity if the rootFragment is other than EventFragment it will assume that it is Main page which will be wrong in all other cases like TicketFragment, AttendeeFragment, OrderDetailsFragment, it was calling EventFragment finish method which was closing our app , so add conditions for these Fragments too.

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/MainActivity.kt [hunk 1]
```diff
@@ -88,10 +91,12 @@ class MainActivity : AppCompatActivity() {
...
         val rootFragment = this.supportFragmentManager.findFragmentById(R.id.rootLayout)
-        if (rootFragment is EventDetailsFragment)
-            super.onBackPressed()
-        else
-            when (currentFragment) {
+        when (rootFragment) {
+            is EventDetailsFragment -> super.onBackPressed()
+            is TicketsFragment -> super.onBackPressed()
+            is AttendeeFragment -> super.onBackPressed()
+            is OrderDetailsFragment -> super.onBackPressed()
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/MainActivity.kt]: you can call `super.onBackPressed()` once like - 
```java
 else -> when (currentFragment) {
                 is SearchFragment,
                 is FavoriteFragment,
                 is OrdersUnderUserFragment,
                 is EventsFragment -> finish()
```

File: app/src/main/java/org/fossasia/openevent/general/MainActivity.kt [hunk 2]
```diff
@@ -88,10 +91,12 @@ class MainActivity : AppCompatActivity() {
...
-        if (rootFragment is EventDetailsFragment)
-            super.onBackPressed()
-        else
-            when (currentFragment) {
+        when (rootFragment) {
+            is EventDetailsFragment,
+            is TicketsFragment,
+            is AttendeeFragment,
+            is OrderDetailsFragment -> super.onBackPressed()
+            else -> when (currentFragment) {
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/MainActivity.kt]: Both when conditions can be written separately. Let mentors review it.

Comment (pranavpandey1998official) [reply app/src/main/java/org/fossasia/openevent/general/MainActivity.kt]: @iamareebjamal how about this the rootFragment is null when we are in the main page so

      if (rootFragment == null) {
                 when (currentFragment) {
                    is SearchFragment,
                    is FavoriteFragment,
                    is OrdersUnderUserFragment,
                    is ProfileFragment -> {
                        loadFragment(supportFragmentManager, EventsFragment(), frameContainer.id)
                        navigation.selectedItemId = navigation_events
                    }
                    is EventsFragment -> finish()
                    else -> super.onBackPressed()
                }
            }
        else super.onBackPressed()

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/MainActivity.kt]: Seems better but reverse the condition and use early return

Comment (pranavpandey1998official) [reply app/src/main/java/org/fossasia/openevent/general/MainActivity.kt]: @iamareebjamal like this 


    if (rootFragment != null) {
                super.onBackPressed()
                return
            }
        when (currentFragment) {
            is SearchFragment,
            is FavoriteFragment,
            is OrdersUnderUserFragment,
            is ProfileFragment -> {
                loadFragment(supportFragmentManager, EventsFragment(), frameContainer.id)
                navigation.selectedItemId = navigation_events
            }
            is EventsFragment -> finish()
            else -> super.onBackPressed()
        }
    }

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/MainActivity.kt]: Yes

**Comment by liveHarshit** [comment]
@pranavpandey1998official This issue is solved in PR #823  with navigation. Sorry for it.

**Pull request closed without merging**